examples/TestAppUwp/NodeDssSignaler.cs

```diff
@@ -289,110 +291,119 @@ public Task SendMessageAsync(Message message)
+                        };
+                        Message msg = JsonConvert.DeserializeObject<Message>(jsonMsg, jsonSettings);
+                        if (msg != null)
+                        {
+                            OnMessage?.Invoke(msg);
+                        }
+                        else
                         {
-                            if (readTask.Exception != null)
-                            {
-                                OnFailure?.Invoke(readTask.Exception);
-                                return;
-                            }
-
-                            token.ThrowIfCancellationRequested();
-
-                            var jsonMsg = readTask.Result;
-                            var msg = JsonConvert.DeserializeObject<Message>(jsonMsg);
-                            if (msg != null)
-                            {
-                                OnMessage?.Invoke(msg);
-                            }
-                            else
-                            {
-                                OnFailure?.Invoke(new Exception($"Failed to deserialize SignalerMessage object from JSON."));
-                            }
-                        });
+                            throw new Exception("Failed to deserialize signaler message from JSON.");
+                        }
                     }
 
-                    // Some time may have passed waiting for GET content; check token again for responsiveness
-                    token.ThrowIfCancellationRequested();
+                    masterToken.ThrowIfCancellationRequested();
 
                     // Repeat task to continue polling
                     long curTime = DateTime.UtcNow.Ticks;
                     long deltaTicks = curTime - lastPollTimeTicks;
                     if (deltaTicks >= pollTimeTicks)
                     {
                         // Previous GET task took more time than polling delay, execute ASAP
-                        Task.Run(nextAction, token);
+                        Task.Run(PollServerOnce, masterToken);
                     }
                     else
                     {
                         // Previous GET task took less time than polling delay, schedule next polling
                         long remainTicks = pollTimeTicks - deltaTicks;
                         int nextScheduleTimeMs = (int)(new TimeSpan(remainTicks).TotalMilliseconds);
-                        Task.Delay(nextScheduleTimeMs, token).ContinueWith(_ => nextAction(), token);
+                        Task.Delay(nextScheduleTimeMs, masterToken).ContinueWith(_ => PollServerOnce(), masterToken);
                     }
-                });
-            };
+                }
+                catch (Exception ex)
+                {
+                    OnFailure?.Invoke(ex);
+                    masterTokenSource.Cancel();
```

Comment:
AFAICS we shouldn't need to cancel `masterTokenSource` since it's only used in this loop which is terminating anyway.

Reply:
Right, it is more obvious now that it's rewritten as an actual loop. I thought we could have a pending task scheduled at that point, but thinking about it more I think you're right that's impossible.

Reply:
Fixed.